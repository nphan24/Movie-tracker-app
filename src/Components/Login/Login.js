import React from 'react';
import { NewUser } from '../../Containers/NewUser/NewUser';
import { SignIn } from '../../Containers/SignIn/SignIn';

export const Login = () => {
  return (
    <article>
      <NewUser />
      <SignIn />
    </article>
  )
}
