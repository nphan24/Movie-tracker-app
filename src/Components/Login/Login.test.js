import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';
import { createLocation } from 'history';

describe('Login', () => {
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot at /new-user', () => {
    const location = createLocation('/new-user');
    const wrapper = shallow(<Login location={location}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot at /signIn', () => {
    const location = createLocation('/signIn');
    const wrapper = shallow(<Login location={location}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
