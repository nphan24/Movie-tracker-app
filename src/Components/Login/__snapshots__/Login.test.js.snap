// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                exact={true}
                path="/new-user"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/signIn"
                render={[Function]}
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/new-user"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/signIn"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/new-user",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/signIn",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/new-user"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/signIn"
                    render={[Function]}
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/new-user"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/signIn"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/new-user",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signIn",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
